x-pack/filebeat/input/httpjson/request_chain_helper.go

```diff
@@ -110,22 +125,25 @@ func fetchValueFromContext(trCtx *transformContext, expression string) (string,
 		}
 	case firstResponse:
 		// since first response body is already a map, we do not need to transform it
-		respMap, err := responseToMap(trCtx.firstResponse, false)
+		respMap, err := responseToMap(trCtx.firstResponse)
 		if err != nil {
 			return "", false, err
 		}
 		val, err = iterateRecursive(respMap, keys[1:], 0)
 		if err != nil {
 			return "", false, err
 		}
+	// In this scenario we treat the expression as a hardcoded value, with which we will replace the fixed-pattern
+	case expression:
+		return expression, true, nil
```

Comment:
What is the intended behaviour when `expression` contains a dot?

Reply:
This case is only matched if the expression does not contain any dot's or if there is only 1 dot as a prefix.

Reply:
Yes, but I don't think that is documented and I can see situations where that would stop users from doing valid things.